Duplicati/Library/Utility/FileBackedList.cs

```diff
@@ -36,13 +36,13 @@ public abstract class FileBackedList<T> : IEnumerable<T>, IDisposable
     {
         private class StreamEnumerator : IEnumerator<T>, System.Collections.IEnumerator
         {
-            private Stream m_stream;
+            private Stream? m_stream;

```

Comment:
The stream can actually not be null and should be readonly.
The reason why it appears nullable is that it is being used as a `m_isDisposed` flag when null, which is bad practice.

Reply:
marking this stream as readonly means it cannot be reassigned in this Dispose methosd
<img width="362" alt="image" src="https://github.com/duplicati/duplicati/assets/60369677/1920a11a-e050-4d83-982c-f7f2267782b1">